Content.Server/Ninja/Systems/SpaceNinjaSystem.cs

```diff
@@ -0,0 +1,354 @@
+    [Dependency] private readonly SharedSubdermalImplantSystem _implants = default!;
+    [Dependency] private readonly SharedTransformSystem _transform = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<SpaceNinjaComponent, ComponentStartup>(OnNinjaStartup);
+        SubscribeLocalEvent<SpaceNinjaComponent, MindAddedMessage>(OnNinjaMindAdded);
+        SubscribeLocalEvent<SpaceNinjaComponent, EmaggedSomethingEvent>(OnDoorjack);
+    }
+
+    public override void Update(float frameTime)
+    {
+        var query = EntityQueryEnumerator<SpaceNinjaComponent>();
+        while (query.MoveNext(out var uid, out var ninja))
+        {
+            UpdateNinja(uid, ninja, frameTime);
+        }
+    }
+
+    /// <summary>
+    /// Turns the player into a space ninja
+    /// </summary>
+    public void MakeNinja(Mind.Mind mind)
+    {
+        if (mind.OwnedEntity == null)
+            return;
+
+        // prevent double ninja'ing
+        var user = mind.OwnedEntity.Value;
+        if (HasComp<SpaceNinjaComponent>(user))
+            return;
+
+        AddComp<SpaceNinjaComponent>(user);
+        SetOutfitCommand.SetOutfit(user, "SpaceNinjaGear", EntityManager);
+        GreetNinja(mind);
+    }
+
+    /// <summary>
+    /// Download the given set of nodes, returning how many new nodes were downloaded.'
+    /// </summary>
+    public int Download(EntityUid uid, List<string> ids)
+    {
+        if (!GetNinjaRole(uid, out var role))
+            return 0;
+
+        var oldCount = role.DownloadedNodes.Count;
+        role.DownloadedNodes.UnionWith(ids);
+        var newCount = role.DownloadedNodes.Count;
+        return newCount - oldCount;
+    }
+
+    /// <summary>
+    /// Gets a ninja's role using the player's mind
+    /// </summary>
+    public static bool GetNinjaRole(Mind.Mind? mind, [NotNullWhen(true)] out NinjaRole? role)
+    {
+        if (mind == null)
+        {
+            role = null;
+            return false;
+        }
+
+        role = (NinjaRole?) mind.AllRoles
+            .Where(r => r is NinjaRole)
+            .FirstOrDefault();
+        return role != null;
+    }
+
+    /// <summary>
+    /// Gets a ninja's role using the player's entity id
+    /// </summary>
+    public bool GetNinjaRole(EntityUid uid, [NotNullWhen(true)] out NinjaRole? role)
+    {
+        role = null;
+        if (!TryComp<MindContainerComponent>(uid, out var mind))
+            return false;
+
+        return GetNinjaRole(mind.Mind, out role);
+    }
+
+    /// <summary>
+    /// Returns the global ninja gamerule config
+    /// </summary>
+    public NinjaRuleComponent RuleConfig()
+    {
+        var rule = EntityQuery<NinjaRuleComponent>().FirstOrDefault();
+        if (rule != null)
+            return rule;
+
+        _gameTicker.StartGameRule("Ninja", out var ruleEntity);
+        return Comp<NinjaRuleComponent>(ruleEntity);
```

Comment:
i dont think this saves a lot of lines, and it shouldnt start the gamerule if the event hasnt started, since that behavior doesnt make sense in the usages i have seen

Reply:
it starts the rule since you can antag ctrl without the event